openedx/adg/lms/applications/admin.py

```diff
@@ -72,3 +81,284 @@ class BusinessLineAdmin(admin.ModelAdmin):
+
+    def date_received(self, obj):
+        return obj.created.strftime('%m/%d/%Y')
+    date_received.short_description = 'Date Received (MM/DD/YYYY)'
+
+    def changelist_view(self, request, extra_context=None):
+        if 'status__exact' not in request.GET:
+            extra_context = {'title': 'APPLICATIONS'}
+        elif request.GET['status__exact'] == 'open':
+            extra_context = {'title': 'OPEN APPLICATIONS'}
+        elif request.GET['status__exact'] == 'accepted':
+            extra_context = {'title': 'ACCEPTED APPLICATIONS'}
+        elif request.GET['status__exact'] == 'waitlist':
+            extra_context = {'title': 'WAITLISTED APPLICATIONS'}
+
+        return super(UserApplicationADGAdmin, self).changelist_view(request, extra_context=extra_context)
+
+    def changeform_view(self, request, object_id=None, form_url='', extra_context=None):
+        application = UserApplication.objects.get(id=object_id)
+        if request.method == 'POST':
+            if 'status' in request.POST:
+                new_status = request.POST.get('status')
+                application.status = new_status
+                if 'internal_note' in request.POST:
+                    note = request.POST.get('internal_note')
+                    application.internal_admin_note = note
+                application.reviewed_by = request.user
+                application.save()
+
+        user_id = UserApplication.objects.get(id=object_id).user.id
+        name_of_applicant = User.objects.get(id=user_id).get_full_name()
+
+        reviewed_by = None
+        review_date = None
+        if application.status != 'open':
+            reviewed_by = application.reviewed_by.get_full_name()
+            review_date = application.modified.strftime('%B %d, %Y')
+
+        extra_context = {
+            'title': name_of_applicant,
+            'adg_view': True,
+            'status': application.status,
+            'reviewer': reviewed_by,
+            'review_date': review_date,
+            'note_for_applicant': application.internal_admin_note
+        }
+        return super(UserApplicationADGAdmin, self).changeform_view(
+            request, object_id, form_url, extra_context=extra_context
+        )
+
+    def email(self, obj):
+        return format_html('<a href="mailto:{email_address}">{email_address}</a>', email_address=obj.user.email)
+
+    def location(self, obj):
+        user_profile = UserProfile.objects.get(user=obj.user)
+        return '{city}, {country}'.format(city=user_profile.city, country=user_profile.country)
+
+    def linkedIn_profile(self, obj):
+        return format_html('<a href={url}>{url}</a>', url=obj.linkedin_url)
+
+    def is_saudi_national(self, obj):
+        extended_user_profile = ExtendedUserProfile.objects.get(user=obj.user)
+        if extended_user_profile.is_saudi_national:
+            return 'Yes'
+        return 'No'
+    is_saudi_national.short_description = 'Are you a Saudi National?'
```

Comment:
We can use verbose_name instead of re-defining and it's localized too.

Reply:
The verbose name won't be effective here. Defining short description is necessary here because `is_saudi_national` does not belong to the user application. Instead we have created a custom field function here, which uses the `is_saudi_national` from the ExtendedUserProfile model. If we don't add short description, the field label rendered in the admin would be "Is saudi national?" i.e. it auto converts the function name to label, by removing underscores.

But you're right that we're re-using the same string in two different places. So I'll create a `constants.py` file in `openedg/adg/lms` and place the string 'Are you a Saudi National?' there so that it can be accessed both by `registration_extension` and `applications` apps.